lib/numerical_analysis/find_peaks.dart

```diff
@@ -0,0 +1,106 @@
+import '../analysis_page.dart';
+
+List<LinearData> FindPeaks(Map<double, double> data, double minProminence, double maxProminence){
+
+  List<LinearData> dataConverted = [];
+  List<LinearData> peaks = [];
+  for (double key in data.keys) {
+    dataConverted.add(LinearData(key, data[key]!));
+  }
+
+  for(int i = 0; i < data.length; i++) {
+      //edge cases
+      if(i == 0){
```

Comment:
Could we add one LinearData at the start and one at the end, both with value of 0 and just get rid of those ifs? Just one `for(int i = 1; i < data.length - ;i++){[...]} `